File: ThreeDSecure/src/test/java/com/braintreepayments/api/ThreeDSecureV2LabelCustomizationUnitTest.java
package com.braintreepayments.api;

import android.os.Parcel;

import com.cardinalcommerce.shared.userinterfaces.LabelCustomization;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class ThreeDSecureV2LabelCustomizationUnitTest {

    @Test
    public void setsAllCardinalClassProperties() {
        ThreeDSecureV2LabelCustomization sut = new ThreeDSecureV2LabelCustomization();
        sut.headingTextColor("#00FF00");
        sut.headingTextFontName("Comic Sans");
        sut.headingTextFontSize(12);
        sut.textColor("#ff0000");
        sut.textFontSize(19);
        sut.textFontName("Arial");

        LabelCustomization cardinalLabelCustomization = sut.getCardinalLabelCustomization();
        assertEquals("#00FF00", cardinalLabelCustomization.getHeadingTextColor());
        assertEquals("Comic Sans", cardinalLabelCustomization.getHeadingTextFontName());
        assertEquals(12, cardinalLabelCustomization.getHeadingTextFontSize());
        assertEquals("#ff0000", cardinalLabelCustomization.getTextColor());
        assertEquals(19, cardinalLabelCustomization.getTextFontSize());
        assertEquals("Arial", cardinalLabelCustomization.getTextFontName());
    }

    @Test
    public void writeToParcel() {
        ThreeDSecureV2LabelCustomization customization = new ThreeDSecureV2LabelCustomization();
        customization.headingTextColor("#FFFFFF");
        customization.headingTextFontName("Times New Roman");
        customization.headingTextFontSize(30);
        customization.textColor("#121212");
        customization.textFontName("Helvetica");
        customization.textFontSize(15);

        Parcel parcel = Parcel.obtain();
        customization.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);

        ThreeDSecureV2LabelCustomization actual = (ThreeDSecureV2LabelCustomization) ThreeDSecureV2LabelCustomization.CREATOR.createFromParcel(parcel);

        assertEquals("#FFFFFF", actual.getHeadingTextColor());
        assertEquals("Times New Roman", actual.getHeadingTextFontName());
        assertEquals(30, actual.getHeadingTextFontSize());
        assertEquals("#121212", actual.getTextColor());
        assertEquals("Helvetica", actual.getTextFontName());
        assertEquals(15, actual.getTextFontSize());
    }
}

File: ThreeDSecure/src/test/java/com/braintreepayments/api/threedsecure/ThreeDSecureV2LabelCustomizationUnitTest.kt
package com.braintreepayments.api.threedsecure

import android.os.Parcel
import junit.framework.TestCase.assertEquals
import kotlinx.parcelize.parcelableCreator
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class ThreeDSecureV2LabelCustomizationUnitTest {

    @Test
    fun `sets all cardinalClass properties correctly`() {
        val sut = ThreeDSecureV2LabelCustomization(
            headingTextColor = "#00FF00",
            headingTextFontName = "Comic Sans",
            headingTextFontSize = 12,
            textColor = "#ff0000",
            textFontName = "Arial",
            textFontSize = 19
        )

        val cardinalLabelCustomization = sut.cardinalLabelCustomization

        assertEquals("#00FF00", cardinalLabelCustomization.headingTextColor)
        assertEquals("Comic Sans", cardinalLabelCustomization.headingTextFontName)
        assertEquals(12, cardinalLabelCustomization.headingTextFontSize)
        assertEquals("#ff0000", cardinalLabelCustomization.textColor)
        assertEquals(19, cardinalLabelCustomization.textFontSize)
        assertEquals("Arial", cardinalLabelCustomization.textFontName)
    }

    @Test
    fun `creates ThreeDSecureV2LabelCustomization and parcels it correctly`() {
        val customization = ThreeDSecureV2LabelCustomization(
            headingTextColor = "#FFFFFF",
            headingTextFontName = "Times New Roman",
            headingTextFontSize = 30,
            textColor = "#121212",
            textFontName = "Helvetica",
            textFontSize = 15
        )

        val parcel = Parcel.obtain().apply {
            customization.writeToParcel(this, 0)
            setDataPosition(0)
        }

        val actual = parcelableCreator<ThreeDSecureV2LabelCustomization>().createFromParcel(parcel)
        assertEquals("#FFFFFF", actual.headingTextColor)
        assertEquals("Times New Roman", actual.headingTextFontName)
        assertEquals(30, actual.headingTextFontSize)
        assertEquals("#121212", actual.textColor)
        assertEquals(15, actual.textFontSize)
        assertEquals("Helvetica", actual.textFontName)
    }
}
